### DISCUSSION TITLE
How to get the proper `StoreApi<S>` type with the modified `subscribe` from `subscribeWithSelector`

### DISCUSSION THREAD

**Discussion by jsardev** [pmndrs/zustand#2192] [Q&A]
I want to create a helper function that is converting a subscribed store value to an Observable. Unfortunately, I can not find any exported type for the modified `StoreApi` type after adding the `subscribeWithSelector` middleware.

For now, I just copy-pasted the type from [zustand source code](https://github.com/pmndrs/zustand/blob/a836356b5e4062abde852af3ba5ee6fc3f1a41fe/src/middleware/subscribeWithSelector.ts#L28-L40).

To give more context, here's the helper function:
```ts
type ToStreamOptions = {
  fireImmediately?: boolean
}

export const toStream = <TState, TValue>(
  store: StoreApi<TState> & StoreSubscribeWithSelector<TState>,
  selector: (value: TState) => TValue,
  { fireImmediately }: ToStreamOptions = {},
): Observable<TValue> => {
  return new Observable<TValue>((subscriber) => {
    if (fireImmediately) {
      subscriber.next(selector(store.getState()))
    }
    const unsubscribe = store.subscribe(selector, (value) => {
      subscriber.next(value)
    })
    return () => unsubscribe()
  })
}
```

Is there any way I could infer the type of the passed store? If not, maybe I'm unaware of some types I could use that are actually exported by `zustand`?

Any help would be very appreciated 🙏

**Comment by dbritto-dev** [comment]
@sarneeh would you mind sharing us a minimal repro on codesandbox?

**Reply by jsardev** [reply]
@dbritto-dev Sure: https://codesandbox.io/s/zustand-type-issue-xhmgmv?file=/src/toStream.ts
